DEVELOPING.md

```diff
@@ -105,6 +111,14 @@ To watch them actually run interactively, you can use `yarn cypress open`. Be aw
 
 If your tests are failing with a message about "deadlock something whatever", do `yarn test --run-in-band`. This makes the tests run sequentially.
 
+If `yarn test` is not running all of the tests, navigate to `server/test` folder and run `yarn jest --config ./test/jest-integration.json -i --run-in-band` if you would like to run all the tests. To run the tests of a specific file (e.g. course.integration.ts), you can use `yarn jest --config ./test/jest-integration.json -i --run-in-band course` 
+
+{% note %}
```

Comment:
I am not really familiar with this syntax, what does it do?

Reply:
so when i was debugging why half the tests weren't running, i found they kept failing on specifically that command (presumably it's set up somewhere to run when you run `yarn test`. However, when I found that I ran specifically the command that wasn't running, it will actually run correctly.

From chatgpt:

--config ./test/jest-integration.json: Specifies the path to the Jest configuration file to be used for the test run. In this case, it's pointing to a configuration file located at ./test/jest-integration.json.

-i or --interactive: Starts Jest in interactive watch mode, enabling you to watch for changes in your source code and rerun tests automatically. It also provides an interactive command prompt for managing tests.

--run-in-band: This option runs all tests serially in the current process rather than parallelizing them. This can be useful for debugging purposes or when running into issues with parallel execution.

Reply:
I was asking about this:

```
{% note %}
```

Reply:
Oh that's a callout. It's rendered in github as a highlighted block section